src/rest/impl/operations.service.ts

```diff
@@ -192,6 +195,19 @@ export default class OperationsServiceImpl extends OperationsService {
     return operationReport;
   }
 
+/**
+ * Updates current year approved contracts to SAP
+ */
+  private updateCurrentYearApprovedContractsToSap = async () => {
+    const operationReport = await models.createOperationReport("UPDATE_CURRENT_YEAR_APPROVED_CONTRACTS_TO_SAP");
+    const contracts = await models.listContractsByStatusAndYear("APPROVED", new Date().getFullYear());
+    for (const contract of contracts) {
+      tasks.enqueueUpdateCurrentYearApprovedContractsToSap(operationReport.id, contract);
```

Comment:
Should this function call have await?

Reply:
No, since this just adds all contracts as separate operations to the operation queue, that executes the operations asynchronously in the background.